### PR TITLE
Ast generation

### PR DESCRIPTION
**Pull request by Toby-Lawrance**

### CODE REVIEW

File: src/parser/inc/LiteralASTNode.hpp [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#pragma once
+
+#include "ExprASTNode.hpp"
+
+namespace AST {
+template <class T>
```
Comment (the-Bruce) [inline src/parser/inc/LiteralASTNode.hpp:6]: Templating... Brave

File: src/parser/samples/testImportUse.cm [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import testSL
+
+func int folder(int acc, int[1] nums){
+    return (head(nums)) + acc
```
Comment (the-Bruce) [inline src/parser/samples/testImportUse.cm:4]: should be:
`return testSL.head(nums)) + acc`

**Review by the-Bruce — CHANGES_REQUESTED** [review]
One slight change to the examples, otherwise that should be fine

**Review by AlfiePowers — DISMISSED** [review]
Looks good to me, check with others

**Review by the-Bruce — APPROVED** [review]
Looks good.

**Pull request merged**